src/suggestionlistworker.cpp

```diff
@@ -48,13 +30,20 @@ void SuggestionListWorker::run()
         QUrl url;
         url.setScheme("zim");
         url.setHost(currentZimId + ".zim");
-        kiwix::SuggestionsList_t suggestions;
-        searchSuggestionsSmart(archive, m_text.toStdString(), 15, suggestions);
-        for (auto& suggestion: suggestions) {
-            QString path = QString("/") + QString::fromStdString(suggestion.getPath());
-            url.setPath(path);
-            suggestionList.append(QString::fromStdString(suggestion.getTitle()));
-            urlList.append(url);
+        int suggestionsCount = 15;
+        auto prefix = m_text.toStdString();
+        auto suggestionSearcher = zim::SuggestionSearcher(*archive);
+        if (archive->hasTitleIndex()) {
```

Comment:
Why limiting suggestion to when archive has a title index ?
If archive has no title index, suggestion may indeed not be the best (as we search in the title listing) but it is better than nothing.